lib/github/event.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module GitHub
+  class Event
+    attr_accessor :event_client
+
+    def initialize(access_token, options = { auto_paginate: false })
+      options[:access_token] = access_token
+
+      @event_client = Octokit::Client.new(options)
+    end
+
+    def latest_push_event(repo_id, options = { per_page: 100, page: 1 })
+      events = event_client.repository_events(repo_id, options).select do |event|
+        event.type == 'PushEvent' ||
+          (event.type == 'CreateEvent' && event.payload.ref.present?)
+      end
+
+      events.length.positive? ? events.first : nil
```

Comment:
Nice catch! Should have used `#empty` for this. 

I'm going to simplify this further using `#first` only as the result will be the same according to the [docs](https://ruby-doc.org/core-2.4.0/Array.html#method-i-first).